service/storage_proxy.cc

```diff
@@ -2137,7 +2137,7 @@ future<> storage_proxy::mutate_begin(unique_response_handler_vector ids, db::con
         auto f = response_wait(response_id, timeout);
         send_to_live_endpoints(protected_response.release(), timeout); // response is now running and it will either complete or timeout
         return f;
-    });
+    }, result<>(bo::success()), utils::result_reducer{});
```

Comment:
May be instead of `utils::result_reducer` you will provide `outcome_parallel_for_each` which hides the map_reduce details inside?

Reply:
Done. I named it `result_parallel_for_each` - the boost outcome library provides both `result` and `outcome` types, so I think a `result_` prefix generates less confusion.